python_modules/libraries/dagster-airbyte/dagster_airbyte/components/workspace_component/component.py

```diff
@@ -0,0 +1,149 @@
+from collections.abc import Sequence
+from functools import cached_property
+from typing import Annotated, Callable, Optional, Union
+
+import dagster as dg
+import dagster.components as dg_components
+import pydantic
+from dagster._core.definitions.asset_check_factories.schema_change_checks import BaseModel
+from dagster._core.definitions.job_definition import default_job_io_manager
+from dagster.components import Component, ComponentLoadContext, Model, Resolvable
+from dagster.components.resolved.base import resolve_fields
+from dagster.components.utils import TranslatorResolvingInfo
+from dagster_shared import check
+from typing_extensions import TypeAlias
+
+from dagster_airbyte.asset_defs import build_airbyte_assets_definitions
+from dagster_airbyte.resources import AirbyteCloudWorkspace
+from dagster_airbyte.translator import (
+    AirbyteConnection,
+    AirbyteConnectionTableProps,
+    DagsterAirbyteTranslator,
+)
+
+
+def resolve_translation(context: dg_components.ResolutionContext, model):
+    info = TranslatorResolvingInfo(
+        "props",
+        asset_attributes=model,
+        resolution_context=context,
+        model_key="translation",
+    )
+    return lambda base_asset_spec, props: info.get_asset_spec(
+        base_asset_spec,
+        {
+            "props": props,
+            "spec": base_asset_spec,
+        },
+    )
+
+
+TranslationFn: TypeAlias = Callable[[dg.AssetSpec, AirbyteConnectionTableProps], dg.AssetSpec]
+ResolvedTranslationFn: TypeAlias = Annotated[
+    TranslationFn,
+    dg_components.Resolver(
+        resolve_translation,
+        model_field_type=Union[str, dg_components.AssetAttributesModel],  # type: ignore
+    ),
+]
+
+
+class ProxyDagsterAirbyteTranslator(DagsterAirbyteTranslator):
+    def __init__(self, fn: TranslationFn):
+        self.fn = fn
+
+    def get_asset_spec(self, props: AirbyteConnectionTableProps) -> dg.AssetSpec:
+        base_asset_spec = super().get_asset_spec(props)
+        spec = self.fn(base_asset_spec, props)
+
+        return spec
+
+
+class AirbyteCloudWorkspaceModel(BaseModel):
+    workspace_id: str = pydantic.Field(..., description="The Airbyte Cloud workspace ID.")
```

Comment:
`Annotated` instead? I guess it doesn't matter but now that ive seen how the sausage is made i know the `Annotated` impl is less greasy